agent/src/beerocks/slave/agent_db.h

```diff
@@ -0,0 +1,78 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2019-2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include <iostream>
+#include <memory>
+#include <mutex>
+
+namespace beerocks {
+
+/**
+ * @brief AgentDB is a class used by all main Agent threads on the Agent process to store the Agent
+ * common data.
+ * The class is implemented using singleton design pattern.
+ * It is thread safe and being locked and released automatically.
+ * 
+ * How to use:
+ * 
+ * auto db = AgentDB::get();            // Get DB and automatically lock it. The lock will be
+ *                                      // Released when 'db' will be destructed.
+ * 
+ * db->foo = 42;                        // Change database member 'foo' to value 42.
+ * db->bar = "prpl"                     // Set another database member.
+ * auto &my_foo = db->foo;              // Get a refernce to a member.
+ * 
+ * AgentDB::get()->foo = 44;            // Set database member directly.
```

Comment:
```suggestion
 * @code
 * auto db = AgentDB::get();            // Get DB and automatically lock it. The lock will be
 *                                      // Released when 'db' will be destructed.
 * 
 * db->foo = 42;                        // Change database member 'foo' to value 42.
 * db->bar = "prpl"                     // Set another database member.
 * auto &my_foo = db->foo;              // Get a refernce to a member.
 * 
 * AgentDB::get()->foo = 44;            // Set database member directly.
 * @endcode
```
(same with the other examples).